--- mysql.php.orig	2017-09-29 11:14:23.000000000 -0700
+++ mysql.php	2017-10-17 16:13:21.106860715 -0700
@@ -114,7 +114,7 @@
   function import_dump($dump_file, $creds) {
     extract($creds);
 
-    $cmd = sprintf("mysql --defaults-file=/dev/fd/3 %s", escapeshellcmd($db_name));
+    $cmd = sprintf("mysql --defaults-extra-file=/dev/fd/3 %s", escapeshellcmd($db_name));
 
     $success = $this->safe_shell_exec($cmd, $db_host, $db_user, $db_passwd, $dump_file);
 
@@ -260,7 +260,7 @@
     } // else
 
     // Mixed copy-paste of drush_shell_exec and provision_shell_exec.
-    $cmd = sprintf("mysqldump --defaults-file=/dev/fd/3 %s --single-transaction --quick --no-autocommit %s", $gtid_option, escapeshellcmd(drush_get_option('db_name')));
+    $cmd = sprintf("mysqldump --defaults-extra-file=/dev/fd/3 %s --single-transaction --quick --no-autocommit %s", $gtid_option, escapeshellcmd(drush_get_option('db_name')));
 
     // Fail if db file already exists.
     $dump_file = fopen(d()->site_path . '/database.sql', 'x');
