--- server.tpl.php.orig	2018-05-02 06:57:50.000000000 -0700
+++ server.tpl.php	2018-07-27 15:36:44.768802852 -0700
@@ -1,15 +1,7 @@
 # Aegir web server configuration file
 
-NameVirtualHost *:<?php print $http_port; ?>
-
-
 <VirtualHost *:<?php print $http_port; ?>>
   ServerName default
-
-  <IfModule mod_rewrite.c>
-    RewriteEngine on
-    RewriteRule ^(?!(/\.well-known/acme-challenge/.+)) - [R=404,L,NC]
-  </IfModule>
 </VirtualHost>
 
 
